### DISCUSSION TITLE
Please add to README how to build additional packaged like dune-configurator

### DISCUSSION THREAD

**Discussion by yurivict** [ocaml/dune#5978] [Q&A]
README says:
```
$ ocaml bootstrap.ml
$ ./dune.exe build -p dune --profile dune-bootstrap
$ ./dune.exe install dune
```

But for many projects dune needs to be built with other packages.

Arch linux builds like this:
```
make dune.exe
./dune.exe build -p 'dune,dune-action-plugin,dune-build-info,dune-configurator,dune-glob,dune-private-libs,dune-site' --profile dune-bootstrap
```
Is this correct? Is the list of packages complete?

See the Arch package here: https://github.com/archlinux/svntogit-community/blob/packages/dune/trunk/PKGBUILD

Why are these additional packages bundled with dune? Shouldn't they be separate packages?

**Comment by cwong-ocaml** [comment]
The extra exported libraries are interfaces for external applications to configure Dune. As they are layers over some of Dune's internals, it doesn't make sense to distribute them without Dune itself (or to distribute Dune without them -- they're linked into the Dune binary anyway).

As far as I know, we don't maintain the Arch Linux package (at least, none of the names of the maintainers of that package are familiar to me), so you'll have to ask them why it's built the way it is.

EDIT: Confirmed, we don't maintain that package. The two supported ways of installing Dune are from source as detailed in the README or via opam.

**Comment by yurivict** [comment]
> [...] it doesn't make sense to distribute them without Dune itself [...]

When dune is built according to the above instructions (```./dune.exe build -p dune --profile dune-bootstrap```) -- it builds without additional libraries. If this doesn't make sense, why doesn't the published default way build them?

There is a circular dependency problem: csexp depends on dune, and dune with additional libraries depends on csexp.
So dune the libraries can't be built without first building it without libraries.

**Comment by cwong-ocaml** [comment]
> it builds without additional libraries.

You are conflating "building" with "packaging". By specifying `-p dune,dune-configurator`, you are asking Dune to build all stanzas in the `dune-configurator` package, many of which are not necessary to produce the `dune.exe` binary. In contrast, `-p dune` will only build `dune.exe`.

> There is a circular dependency problem: csexp depends on dune, and dune with additional libraries depends on csexp.

The `csexp` package used by Dune is vendored, which you can find in `vendor/csexp`.

**Comment by yurivict** [comment]
> The csexp package used by Dune is vendored, which you can find in vendor/csexp.

These build commands fail:
```
ocaml bootstrap.ml && ./dune.exe build -p dune,dune-action-plugin,dune-build-info,dune-configurator,dune-glob,dune-private-libs,dune-site --profile dune-bootstrap
```

```
File "otherlibs/configurator/src/dune", line 7, characters 17-22:
7 |  (libraries unix csexp result)
                     ^^^^^
Error: Library "csexp" not found.
```

Somehow it fails to pick up the vendored ```csexp```.

**Reply by rgrinberg** [reply]
All the additional packages in the dune repo have opam files with their build instructions.
